leaflet-osm.js

```diff
@@ -232,101 +232,74 @@ L.OSM.DataLayer = L.FeatureGroup.extend({
 });
 
 L.OSM.XMLParser = {
-  getChangesets: function (xml) {
-    var result = [];
-
-    var nodes = xml.getElementsByTagName("changeset");
-    for (var i = 0; i < nodes.length; i++) {
-      var node = nodes[i], id = node.getAttribute("id");
-      result.push({
-        id: id,
-        type: "changeset",
-        latLngBounds: L.latLngBounds(
-          [node.getAttribute("min_lat"), node.getAttribute("min_lon")],
-          [node.getAttribute("max_lat"), node.getAttribute("max_lon")]),
-        tags: this.getTags(node)
-      });
-    }
-
-    return result;
+  getChangesets(xml) {
+    const changesets = [...xml.getElementsByTagName("changeset")];
+    return changesets.map(cs => ({
```

Comment:
I think you can avoid copying the elements into an array here and just call map on the iterator directly? Something like:

```js
return xml.getElementsByTagName("changeset").map(cs => ({
```

Much the same would apply to a pretty much all of places where an array is being constructed from an iterator.

Reply:
`map` on iterators is a [very recent addition](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator/map).

Reply:
Fair point. I guess we can forget this then.